src/commands/pickFuncProcess.ts

```diff
@@ -73,69 +73,87 @@ async function startFuncTask(context: IActionContext, workspaceFolder: vscode.Wo
             }
 
             const taskInfo: IRunningFuncTask | undefined = runningFuncTaskMap.get(workspaceFolder);
-            // Ensure func task runs for at least 1 second in case it fails on startup and thus `taskError` will be thrown instead
-            if (taskInfo && Date.now() > taskInfo.startTime + 1000) {
-                return taskInfo.processId.toString();
+            if (taskInfo) {
+                try {
+                    // wait for status url to indicate functions host is running
+                    await client.sendRequest(statusRequest);
+                    return taskInfo.processId.toString();
+                } catch {
+                    // ignore
+                }
             }
 
-            await delay(500);
+            await delay(intervalMs);
         }
 
-        throw timeoutError;
+        throw new Error(localize('failedToFindFuncHost', 'Failed to detect running Functions host within "{0}" seconds. You may want to adjust the "{1}" setting.', timeoutInSeconds, `${ext.prefix}.${settingKey}`));
     } finally {
         errorListener.dispose();
     }
 }
 
+function getStatusRequest(funcTask: vscode.Task, intervalMs: number): WebResource {
+    let port: string = '7071';
+    if (typeof funcTask.definition.command === 'string') {
+        const match: RegExpMatchArray | null = funcTask.definition.command.match(/\s+(?:"|'|)(?:-p|--port)(?:"|'|)\s+(?:"|'|)([0-9]+)/i);
+        if (match) {
+            port = match[1];
+        }
+    }
+
+    let request: WebResource = new WebResource();
+    request = request.prepare({ url: `http://localhost:${port}/admin/host/status`, method: 'GET' });
+    request.timeout = intervalMs;
+    return request;
+}
+
+type OSAgnosticProcess = { command: string | undefined; pid: number | string };
+
 /**
- * Gets the innermost child pid for a unix process. This is only really necessary if the user installs 'func' with a tool like 'npm', which uses a wrapper around the main func exe
+ * Picks the child process that we want to use. Scenarios to keep in mind:
+ * 1. On Windows, the rootPid is almost always the parent PowerShell process
+ * 2. On Unix, the rootPid may be a wrapper around the main func exe if installed with npm
+ * 3. Starting with the .NET 5 worker, Windows sometimes has an inner process we _don't_ want like 'conhost.exe'
+ * The only processes we should want to attach to are the "func" process itself or a "dotnet" process running a dll, so we will pick the innermost one of those
  */
-async function getInnermostUnixPid(pid: string): Promise<string> {
-    return await new Promise<string>((resolve: (pid: string) => void, reject: (e: Error) => void): void => {
-        unixPsTree(parseInt(pid), (error: Error | undefined, children: unixPsTree.PS[]) => {
+async function pickChildProcess(rootPid: string): Promise<string> {
+    const children: OSAgnosticProcess[] = process.platform === 'win32' ? await getWindowsChildren(rootPid) : await getUnixChildren(rootPid);
```

Comment:
Seems like you were polling the children before with the while loop/delay, but I don't see that behavior anymore.  Is it not necessary?

Reply:
It was replaced by the polling of "admin/host/status" instead